src/vos/vos_io.c

```diff
@@ -332,7 +332,6 @@ vos_dedup_free_bsgl(struct vos_io_context *ioc, unsigned int sgl_idx,
 		if (biov->bi_buf == NULL)
 			goto next;
 
-		D_ASSERT(!BIO_ADDR_IS_DEDUP(&biov->bi_addr));
```

Comment:
Instead of removing this assert, I prefer to add "biov_dup->bi_buf = NULL" when bio_buf_alloc() failed in vos_dedup_dup_sgl().

Reply:
Ah right, that looks much clean ;-) thanks @NiuYawei